clang/lib/Basic/IdentifierTable.cpp

```diff
@@ -250,6 +250,32 @@ static KeywordStatus getKeywordStatus(const LangOptions &LangOpts,
   return CurStatus;
 }
 
+static bool IsKeywordInCpp(unsigned Flags) {
+  while (Flags != 0) {
+    unsigned CurFlag = Flags & ~(Flags - 1);
+    Flags = Flags & ~CurFlag;
+    switch (static_cast<TokenKey>(CurFlag)) {
+    case KEYCXX:
+    case KEYCXX11:
+    case KEYCXX20:
+    case BOOLSUPPORT:
+    case WCHARSUPPORT:
+    case CHAR8SUPPORT:
+      return true;
+    default:
+      break; // Go to the next flag, try again.
+    }
+  }
+  return false;
```

Comment:
I'm a bit confused by this function. Why does this go through all bits in a loop instead of directly checking them? I'd have expected this whole function to be something like `return Flags & (KEYALLCXX|BOOLSUPPORT|WCHARSUPPORT|CHAR8SUPPORT);`

Reply:
Code was lifted from https://github.com/llvm/llvm-project/blob/a4ceac7e3e04c32cb3e334eb89b54d8e99a298f8/clang/lib/Basic/IdentifierTable.cpp#L240 but yeah, this seems to be something I can simplify

Reply:
I updated in 0b2ab11321d2b400987e9ade28a221dff67aea7d thank you for the catch!